ccf_transaction_processor/transaction_processor/contract_registry.h

```diff
@@ -76,13 +76,14 @@ namespace ccf
   {
     string contract_id;
     std::vector<uint8_t> contract_code_hash;
+    std::vector<uint8_t> contract_metadata_hash;
```

Comment:
What does `contract_metadata_hash` cover?

Reply:
hash of the contract id, public signing key and public encryption key.

Reply:
Got it. So the `contract_code_hash` is actually a measurement over the executable code itself? Clarifying since `code_hash` in the contract code object in the eservice does include metadata.

Reply:
The code hash is the hash(hash(nonce) + hash(source) + hash(compilation report)). the separation between code hash and metadata hash is supposed to reflect the difference between behavior (code hash) and identity/communication (metadata hash). So the code hash is supposed to capture what the object does while the metadata hash is supposed to support communication to/from the object.